### PR TITLE
Add DragonFly BSD support

### PR DESCRIPTION
**Pull request by strangelittlemonkey**

### CODE REVIEW

File: news/dragonfly_support.rst [hunk 1]
```diff
@@ -0,0 +1,11 @@
+**Added:** DragonFly BSD support
```
Comment (gforsyth) [inline news/dragonfly_support.rst]: The note needs to be on it's own line, so something like

```
**Added:**

- DragonFly BSD support
```

File: xonsh/aliases.py [hunk 2]
```diff
@@ -13,7 +13,7 @@
 from xonsh.environ import locate_binary, make_args_env
 from xonsh.foreign_shells import foreign_shell_data
 from xonsh.jobs import jobs, fg, bg, clean_jobs
-from xonsh.platform import ON_ANACONDA, ON_DARWIN, ON_WINDOWS, ON_FREEBSD, ON_NETBSD
+from xonsh.platform import ON_ANACONDA, ON_DARWIN, ON_WINDOWS, ON_FREEBSD, ON_NETBSD, ON_DRAGONFLY
```
Comment (gforsyth) [inline xonsh/aliases.py]: Ahh, and I think `black` is unhappy about the length of this line -- can you run `black` on `aliases.py` to fix it?  Thanks!

**Comment by scopatz** [comment]
Thanks for putting this in @strangelittlemonkey!

File: xonsh/aliases.py (hunk 2 above)
Comment (strangelittlemonkey) [reply xonsh/aliases.py]: Eh, so for my OS py36-black pulled in like 4GB of deps and then it was broken and I had to install something else to fix it, got it up, but it seems that the lint also changed more than just that line. If you'd prefer I can take those changes back out, but I left them for the moment. Your call.

Comment (scopatz) [reply xonsh/aliases.py]: Sorry about the black issue, that seems uncharacteristically large. I'll fix the remaining black issue

**Comment by scopatz** [comment]
Thanks @strangelittlemonkey!

**Pull request merged**